src/Host/Client/Impl/BrokerServices/ISessionsWebService.cs

```diff
@@ -2,13 +2,13 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System.Collections.Generic;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.R.Host.Protocol;
 
 namespace Microsoft.R.Host.Client.BrokerServices {
     public interface ISessionsWebService {
-        Task<IEnumerable<SessionInfo>> GetAsync();
-
-        Task<SessionInfo> PutAsync(string id, SessionCreateRequest request);
+        Task<IEnumerable<SessionInfo>> GetAsync(CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
Honestly, I'd rather merge all the separate *`Servise` classes into one and change the names of the methods. @int19h, what do you think?

---
In reply to: [79269882](https://github.com/Microsoft/RTVS/pull/2342#discussion_r79269882) [](ancestors = 79269882)